transform/kusto_to_prometheus.go

```diff
@@ -0,0 +1,333 @@
+package transform
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+// TransformConfig defines how to convert KQL query results to metrics format
+type TransformConfig struct {
+	// MetricNameColumn specifies which column contains the metric name
+	MetricNameColumn string `json:"metricNameColumn,omitempty"`
+
+	// ValueColumn specifies which column contains the metric value
+	ValueColumn string `json:"valueColumn"`
+
+	// TimestampColumn specifies which column contains the timestamp
+	TimestampColumn string `json:"timestampColumn"`
+
+	// LabelColumns specifies columns to use as metric labels
+	LabelColumns []string `json:"labelColumns,omitempty"`
+
+	// DefaultMetricName provides a fallback if MetricNameColumn is not specified
+	DefaultMetricName string `json:"defaultMetricName,omitempty"`
+}
+
+// KustoToPrometheusTransformer transforms KQL query results to Prometheus metrics
+type KustoToPrometheusTransformer struct {
+	config TransformConfig
+	meter  metric.Meter
+}
+
+// MetricData represents a single metric data point
+type MetricData struct {
+	Name      string
+	Value     float64
+	Timestamp time.Time
+	Labels    map[string]string
+}
+
+// NewKustoToPrometheusTransformer creates a new transformer instance
+func NewKustoToPrometheusTransformer(config TransformConfig, meter metric.Meter) *KustoToPrometheusTransformer {
+	return &KustoToPrometheusTransformer{
+		config: config,
+		meter:  meter,
+	}
+}
+
+// Transform converts KQL query results to Prometheus metrics
+func (t *KustoToPrometheusTransformer) Transform(results []map[string]any) ([]MetricData, error) {
+	if len(results) == 0 {
+		return []MetricData{}, nil
+	}
+
+	var metrics []MetricData
+
+	for i, row := range results {
+		metric, err := t.transformRow(row)
+		if err != nil {
+			return nil, fmt.Errorf("failed to transform row %d: %w", i, err)
```

Comment:
Nit: should we consider skipping this row if there is something goofy with it instead of fully bailing? Not sure the right answer.